duecredit/tests/test_api.py

```diff
@@ -41,102 +87,107 @@ def purpose_of_life():
-         def __init__(self):
-             pass
+        # Conception process is usually way too easy to be referenced
+        def __init__(self):
+            pass
 
-         # including functionality within/by the methods
-         @due.dcite('XXX00')
-         def birth(self, gender):
-             return "Rachel was born"
+        # including functionality within/by the methods
+        @due.dcite('XXX00')
+        def birth(self, gender):
+            return "Rachel was born"
 
     kid = Child()
     kid.birth("female")
 
 
-def test_api():
-    yield _test_api, DueCreditCollector()
-    yield _test_api, InactiveDueCreditCollector()
-
-import os
-import sys
-from os.path import dirname, join as pathjoin, pardir, normpath
-from mock import patch
-from subprocess import Popen, PIPE
-
-badlxml_path = pathjoin(dirname(__file__), 'envs', 'nolxml')
-stubbed_script = pathjoin(dirname(__file__), 'envs', 'stubbed', 'script.py')
-
 def run_python_command(cmd=None, script=None):
     """Just a tiny helper which runs command and returns exit code, stdout, stderr"""
-    assert(bool(cmd) != bool(script))  # one or another, not both
+    assert bool(cmd) != bool(script)  # one or another, not both
     args = ['-c', cmd] if cmd else [script]
-    python = Popen([sys.executable] + args, stdout=PIPE, stderr=PIPE)
-    stdout, stderr = python.communicate()  # wait()
-    ret = python.poll()
-    return ret, stdout.decode(), stderr.decode()
-
-mock_env_nolxml = {'PYTHONPATH': "%s:%s" % (badlxml_path, os.environ.get('PYTHONPATH', ''))}
+    try:
+        # run script from some temporary directory so we do not breed .duecredit.p
+        # in current directory
+        tmpdir = tempfile.mkdtemp()
+        python = Popen([sys.executable] + args, stdout=PIPE, stderr=PIPE, cwd=tmpdir)
+        stdout, stderr = python.communicate()  # wait()
+        ret = python.poll()
+    finally:
+        shutil.rmtree(tmpdir)
+    # TODO stdout cannot decode on Windows special character /x introduced
+    return ret, stdout.decode(errors='ignore'), stderr.decode()
 
 
 # Since duecredit and possibly lxml already loaded, let's just test
 # ability to import in absence of lxml via external call to python
-def test_noincorrect_import_if_no_lxml():
+def test_noincorrect_import_if_no_lxml(monkeypatch):
     if on_windows:
-        raise KnownFailure("Fails for some reason on Windows")
-    with patch.dict(os.environ, mock_env_nolxml):
-        # make sure out mocking works here
-        ret, out, err = run_python_command('import lxml')
-        assert_equal(ret, 1)
-        assert_in('ImportError', err)
-        #
-        # make sure out mocking works here
-        ret, out, err = run_python_command('import duecredit')
-        assert_equal(err, '')
-        assert_equal(out, '')
-        assert_equal(ret, 0)
-
-
-def check_noincorrect_import_if_no_lxml_numpy(kwargs, env):
+        pytest.xfail("Fails for some reason on Windows")
```

Comment:
have no windows env handy -- could you please check if it is still the case -- may be we fixed for it already